Patch-mainline: submitted 04aug2009
References: bnc#498708
From: NeilBrown <neilb@suse.de>
Date: Tue, 4 Aug 2009 15:06:39 +1000
Subject: [PATCH 10/12] sunrpc: fix memory leak in unix_gid cache.

When we look up an entry in the uid->gidlist cache, we take
a reference to the content but don't drop the reference to the
cache entry.  So it never gets freed.

Signed-off-by: NeilBrown <neilb@suse.de>

---
 net/sunrpc/svcauth_unix.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.27-SLE11_BRANCH.orig/net/sunrpc/svcauth_unix.c
+++ linux-2.6.27-SLE11_BRANCH/net/sunrpc/svcauth_unix.c
@@ -670,6 +670,7 @@ static int unix_gid_find(uid_t uid, stru
 	case 0:
 		*gip = ug->gi;
 		get_group_info(*gip);
+		cache_put(&ug->h, &unix_gid_cache);
 		return 0;
 	case -ETIMEDOUT:
 		prevug = ug;
